api/envoy/api/v2/route/route_components.proto

```diff
@@ -696,6 +696,27 @@ message RouteAction {
     google.protobuf.BoolValue enabled = 2;
   }
 
+  // Describe how to match the path and rewrite it using a regular expression
```

Comment:
Shall we extract this type into `envoy.type`? I think we're likely to reuse it in other places?

Reply:
Looking through this again, yes, I think it makes sense to move this to a matcher. How about `RegexMatchAndSubstitute` or something like that?

Reply:
Looking through this again, yes, I think it makes sense to move this to a matcher. How about `RegexMatchAndSubstitute` or something like that?

Reply:
I merged from the `master` branch again to resolve some conflicts and made the requested change, renaming the `RegexRewrite` message into `RegexMatchAndSubstitute` and making it a general type so it can potentially be re-used elsewhere.